satellite/console/consoleweb/server.go

```diff
@@ -192,7 +212,282 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// bucketUsageReportHandler generate bucket usage report page for project
+// authMiddlewareHandler performs initial authorization before every request.
+func (server *Server) authMiddlewareHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		var err error
+		defer mon.Task()(&ctx)(&err)
+		token := getToken(r)
+
+		ctx = auth.WithAPIKey(ctx, []byte(token))
+		auth, err := server.service.Authorize(ctx)
+		if err != nil {
+			ctx = console.WithAuthFailure(ctx, err)
+		} else {
+			ctx = console.WithAuth(ctx, auth)
+		}
+
+		handler.ServeHTTP(w, r.Clone(ctx))
+	})
+}
+
+// tokenRequestHandler authenticates User by credentials and returns auth token.
+func (server *Server) tokenRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type tokenRequestModel struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+
+	var tokenRequest tokenRequestModel
+	err = json.NewDecoder(r.Body).Decode(&tokenRequest)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
```

Comment:
Don't use numbers for status codes instead use the constants defined in `http` package https://golang.org/pkg/net/http/#pkg-constants.